src/main/java/core/basesyntax/ColorSupplier.java

```diff
@@ -1,7 +1,14 @@
 package core.basesyntax;
 
-public class ColorSupplier {
+import java.util.Random;
+
+public class ColorSupplier extends Ball {
```

Comment:
The ColorSupplier class should not extend the Ball class. Inheritance implies a 'is-a' relationship, but in this context, a ColorSupplier is not a type of Ball, but a utility to supply colors. Consider removing the extension of Ball class.